lib/ansible/modules/cloud/vmware/vmware_host_facts.py

```diff
@@ -0,0 +1,178 @@
+    return facts
+
+
+def get_datastore_facts(host):
+    facts = {}
+    facts['ansible_datastore'] = []
+    for store in host.datastore:
+        _tmp = {
+            'name': store.summary.name,
+            'total': sizeof_fmt(store.summary.capacity),
+            'free': sizeof_fmt(store.summary.freeSpace)
+        }
+        facts['ansible_datastore'].append(_tmp)
+    return facts
+
+
+def get_network_facts(host):
+    facts = {}
+    facts['ansible_interfaces'] = []
+    facts['ansible_all_ipv4_addresses'] = []
+    for nic in host.config.network.vnic:
+        device = nic.device
+        facts['ansible_interfaces'].append(device)
+        facts['ansible_all_ipv4_addresses'].append(nic.spec.ip.ipAddress)
+        _tmp = {
+            'device': device,
+            'ipv4': {
+                'address': nic.spec.ip.ipAddress,
+                'netmask': nic.spec.ip.subnetMask
+            },
+            'macaddress': nic.spec.mac,
+            'mtu': nic.spec.mtu
+        }
+        facts['ansible_' + device] = _tmp
+    return facts
+
+
+def get_system_facts(host):
+    sn = 'NA'
+    for info in host.hardware.systemInfo.otherIdentifyingInfo:
+        if info.identifierType.key == 'ServiceTag':
+            sn = info.identifierValue
+    facts = {
+        'ansible_distribution': host.config.product.name,
+        'ansible_distribution_version': host.config.product.version,
+        'ansible_distribution_build': host.config.product.build,
+        'ansible_os_type': host.config.product.osType,
+        'ansible_system_vendor': host.hardware.systemInfo.vendor,
+        'ansible_hostname': host.summary.config.name,
+        'ansible_product_name': host.hardware.systemInfo.model,
+        'ansible_product_serial': sn,
+        'ansible_bios_date': host.hardware.biosInfo.releaseDate,
+        'ansible_bios_version': host.hardware.biosInfo.biosVersion
+    }
+    return facts
+
+
+def sizeof_fmt(num):
+    """
+    Returns the human readable version of a file size
+
+    :param num:
+    :return:
+    """
+    for item in ['bytes', 'KB', 'MB', 'GB']:
+        if num < 1024.0:
+            return "%3.1f%s" % (num, item)
+        num /= 1024.0
+    return "%3.1f%s" % (num, 'TB')
+
+
+def all_facts(content):
+    host = find_obj(content, [vim.HostSystem], None)
+    ansible_facts = {}
+    ansible_facts.update(get_cpu_facts(host))
+    ansible_facts.update(get_memory_facts(host))
+    ansible_facts.update(get_datastore_facts(host))
+    ansible_facts.update(get_network_facts(host))
+    ansible_facts.update(get_system_facts(host))
+    return ansible_facts
+
+
+def main():
+
+    argument_spec = vmware_argument_spec()
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+
+    if not HAS_PYVMOMI:
+        module.fail_json(msg='pyvmomi is required for this module')
+
+    try:
+        content = connect_to_api(module)
+        data = all_facts(content)
+        module.exit_json(changed=False, ansible_facts=data)
```

Comment:
We don't like these catch-all statements in Ansible.

In this case I would expect the library to handle any exceptions caused by connect_to_api.
`all_facts()` should be quite safe, if not the smaller pieces that may raise exceptions should be wrapped.
`exit_json()` is not expected to raise exceptions.

So I'd remove the whole try-block here altogether.